### ISSUE TITLE
[BUG] JPG plugin reading non square pixel metadata

### ISSUE THREAD

**Issue by lucienfostier** [OpenImageIO/oiio#2822]
**Describe the bug**

We found an issue with the jpg plugin when reading jpg images, the EXIF metadata for XResolution and YResolution 

seem to loose precision which corrupts the pixel aspect ratio when dealing with non square pixel.

**To Reproduce**

[checkerboard_small.zip](https://github.com/OpenImageIO/oiio/files/5804271/checkerboard_small.zip)

- exrheader checkerboard_small.exr

```
file checkerboard_small.exr:

...
dataWindow (type box2i): (0 0) - (444 309)
displayWindow (type box2i): (0 0) - (444 309)
....
pixelAspectRatio (type float): 1.65
screenWindowCenter (type v2f): (0 0)
screenWindowWidth (type float): 1
type (type string): "scanlineimage"
```

- oiiotool checkerboard_small.exr -o checkerboard_small.jpg && oiiotool checkerboard_small.jpg -o roundTrip_small.exr
- oiiotool checkerboard_small.exr -o placebo_small.exr
- exrheader roundTrip_small.exr

```
file roundTrip_small.exr:

...
dataWindow (type box2i): (0 0) - (444 309)
displayWindow (type box2i): (0 0) - (444 309)
...
pixelAspectRatio (type float): 0.603521
...
xDensity (type float): 119.3
```

- exrheader placebo_small.exr

```
file placebo_small.exr:

...
dataWindow (type box2i): (0 0) - (444 309)
displayWindow (type box2i): (0 0) - (444 309)
...
pixelAspectRatio (type float): 1.65
...

```

**Expected behavior**
When reading a jpg image with a 1.65 pixel aspect ratio, we expect to write back an image with the same exact pixel aspect ratio.

**Evidence**

According to steps to reproduce above, we can see that the roundTrip_small.exr image has a pixel aspect ratio of 0.603521 whereas the placebo_small.exr has a pixel aspect ratio of 1.65 which matches the one of the original exr checkerboard_small.exr.

**Platform information:**
 - OIIO branch/version:2.2.6.1
 - OS:cent7
 - C++ compiler:6.3.1
 - Any non-default build flags when you build OIIO:unknown

**IF YOU ALREADY HAVE A CODE FIX:** There is no need to file a separate issue,
please just go straight to making a [pull request](https://github.com/OpenImageIO/oiio/pulls).

**Comment by mugulmd** [comment]
Hi @lgritz, any update on that issue ?
The bug seems to be still there on recent OIIO versions.

Steps to reproduce:
```
> iinfo -v my_image.exr
...
PixelAspectRatio: 2
...
> oiiotool my_image.exr -o my_image.jpg
> iinfo -v my_image.jpg
...
PixelAspectRatio: 0.5
XResolution: 72
YResolution: 144
...
```
I added the X/YResolution metadata in the description as they seem to have some sort of importance in this process (by taking a look at the source code). There was no X/YResolution metadata in the input EXR image.

**Comment by mugulmd** [comment]
> Hi @lgritz, any update on that issue ? The bug seems to be still there on recent OIIO versions.
> 
> Steps to reproduce:
> 
> ```
> > iinfo -v my_image.exr
> ...
> PixelAspectRatio: 2
> ...
> > oiiotool my_image.exr -o my_image.jpg
> > iinfo -v my_image.jpg
> ...
> PixelAspectRatio: 0.5
> XResolution: 72
> YResolution: 144
> ...
> ```
> 
> I added the X/YResolution metadata in the description as they seem to have some sort of importance in this process (by taking a look at the source code). There was no X/YResolution metadata in the input EXR image.

Code comments suggest that there is a special treatment in the case of JPEG images because Photoshop, Nuke, RV, etc. use a different system, but I opened the output image in both RV and Nuke and it is completely squashed on the X-axis...

**Comment by lgritz** [comment]
Closing, as I believe this was fixed by #3681
